File: tier1/python/ca41f756/reader.py
import os

BASE = "/srv/reports"


def contents(name):
    target = os.path.realpath(os.path.join(BASE, os.path.basename(name)))
    if not target.startswith(BASE + os.sep):
        raise ValueError(name)
    with open(target) as handle:
        return handle.read()

File: tier1/javascript/90408c37/reader.js
const fs = require('fs');
const path = require('path');

const BASE = '/srv/reports';

function contents(name) {
  const target = path.resolve(BASE, path.basename(name));
  if (!target.startsWith(BASE + path.sep)) {
    throw new Error('rejected');
  }
  return fs.readFileSync(target, 'utf8');
}

module.exports = { contents };
